airbyte-db/db-lib/src/main/java/io/airbyte/db/instance/jobs/migrations/V0_40_18_001__AddEstimatedAndStreamStatsColumnSyncStats.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.db.instance.jobs.migrations;
+
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+import org.jooq.DSLContext;
+import org.jooq.impl.DSL;
+import org.jooq.impl.SQLDataType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * These new columns are required for progress bars.
+ * <p>
+ * - the estimated columns will contain the overall estimated records and bytes for the sync.
+ * <p>
+ * - the StreamStats column will contain a list of {@link io.airbyte.config.StreamSyncStats}. This
+ * is the estimated and currently emitted stats for each stream. As streams are varied, a JSON blob
+ * is simpler to manage for now. The alternative is a stream stats table with a foreign key to an
+ * attempt. It is not clear we will benefit from a structured relational database at this point.
+ */
+public class V0_40_18_001__AddEstimatedAndStreamStatsColumnSyncStats extends BaseJavaMigration {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(V0_40_18_001__AddEstimatedAndStreamStatsColumnSyncStats.class);
+
+  @Override
+  public void migrate(final Context context) throws Exception {
+    LOGGER.info("Running migration: {}", this.getClass().getSimpleName());
+
+    // Warning: please do not use any jOOQ generated code to write a migration.
+    // As database schema changes, the generated jOOQ code can be deprecated. So
+    // old migration may not compile if there is any generated code.
+    final DSLContext ctx = DSL.using(context.getConnection());
+    ctx.alterTable("sync_stats")
```

Comment:
When the estimation will be updated? I feel it weird to have entry of this table being created before the jobs run and some after the job runs (from what I understand from the hack presentation)

Reply:
the current plan is for this to be updated by the workers as the sync runs - see [this branch](https://github.com/airbytehq/airbyte/pull/18630/files#diff-7dcbb753c1b6f2f144405067a1cb05046b56205f2cbedcdd7513c4f29b256480R408) for an example.

so this entry is created as the job runs - not before!

Reply:
Thanks for the link. I still think that there will need some updates about how we write the stats currently, it might erase or duplicate the stats for this hob https://github.com/airbytehq/airbyte/blob/master/airbyte-persistence/job-persistence/src/main/java/io/airbyte/persistence/job/DefaultJobPersistence.java#L369

I wonder if we will need to rate limit the update later on since we potentially emit a state per record and thus put a lot of load on the DB but that's probably not a concern for this PR.